realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -284,6 +288,16 @@ protected final void copy(ColumnInfo rawSrc, ColumnInfo rawDst) {
     }
 
     @Override
+    public RealmInteger realmGet$columnCounter() {
+        return new ManagedRealmInteger(0);
+    }
+
+    @Override
+    public void realmSet$columnCounter(RealmInteger value) {
+        throw new IllegalStateException("Cannot assign to a managed RealmInteger");
```

Comment:
If the `value` is `null`, we should nullify the field.

And I think that it's better just to set `value.longValue()` if the `value` is not `null` in order to provide consistent behavior because we are doing so in `insertOrUpdate()` https://github.com/realm/realm-java/pull/4748/files#diff-9bbf8752325de6eed4a407ae41857948R1069
And under current implementation, `update()` always throws the exception. https://github.com/realm/realm-java/pull/4748/files#diff-9bbf8752325de6eed4a407ae41857948R1230